dash/include/dash/GlobPtr.h

```diff
@@ -104,12 +110,24 @@ class GlobPtr
    * Allows to instantiate GlobPtr without a valid memory
    * space reference.
    */
-  constexpr GlobPtr(
+  GlobPtr(
     const MemorySpace * mem_space,
-    dart_gptr_t gptr)
-  : _dart_gptr(gptr)
+    dart_gptr_t         gptr,
+    index_type          rindex = 0)
+  : _rbegin_gptr(gptr)
   , _mem_space(reinterpret_cast<const MemorySpace *>(mem_space))
```

Comment:
Ah, the argument type should be `const MemSpaceT`. Assignment and copy from other GlobPtr with different memory space type is essential to convert `GlobPtr<T>` to `GlobPtr <const T>`